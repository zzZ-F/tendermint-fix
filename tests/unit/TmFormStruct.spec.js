import { mount } from "@vue/test-utils"
import htmlBeautify from "html-beautify"
import { TmFormStruct } from "@/index"

describe("TmFormStruct", () => {
  let wrapper

  let propsData = {
    width: "narrow",
    submit: function() {
      // console.log("form successfully submitted")
    }
  }

  beforeEach(() => {
    wrapper = mount(TmFormStruct, {
      propsData,
      slots: {
        default: '<div class="fake-fields"></div>',
        footer: '<div class="fake-footer">Fake Footer</div>',
        title: '<div class="fake-title">Fake Title</div>',
        subtitle: '<div class="fake-subtitle">Fake Subtitle</div>'
      }
    })
  })

  it("has a width from props", () => {
    expect(wrapper.vm.width).toBe("narrow")
  })

  it("has a submit function", () => {
    expect(wrapper.vm.submit).toBeTruthy()
  })

  it("has the expected html structure", () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })

  it("has a default slot", () => {
    expect(wrapper.findAll(".fake-fields").length).toBe(1)
  })

  it("has a footer slot", () => {
    expect(
      wrapper
        .find(".fake-footer")
        .text()
        .trim()
    ).toBe("Fake Footer")
  })

  it("has a title slot", () => {
    expect(
      wrapper
        .find(".fake-title")
        .text()
        .trim()
    ).toBe("Fake Title")
  })

  it("has a subtitle slot", () => {
    expect(
      wrapper
        .find(".fake-subtitle")
        .text()
        .trim()
    ).toBe("Fake Subtitle")
  })
})
