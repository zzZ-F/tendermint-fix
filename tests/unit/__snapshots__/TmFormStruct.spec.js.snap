// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmFormStruct has the expected html structure 1`] = `
<form class="tm-form tm-form-narrow">
  <div class="tm-form-container">
    <header class="tm-form-header">
      <div class="tm-form-title">
        <div class="fake-title">
          Fake Title
        </div>
      </div>
      <div class="tm-form-subtitle">
        <div class="fake-subtitle">
          Fake Subtitle
        </div>
      </div>
    </header>
    <main class="tm-form-main">
      <div class="fake-fields"></div>
    </main>
    <footer class="tm-form-footer">
      <div class="fake-footer">
        Fake Footer
      </div>
    </footer>
  </div>
</form>
`;
